### PR TITLE
Bug 1883196: kube-apiserver: silence own loopback requests before /readyz success

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go [hunk 1]
```diff
@@ -116,7 +116,9 @@ func WithNonReadyRequestLogging(handler http.Handler, hasBeenReadyCh <-chan stru
 		// ignore connections to local IP. Those clients better know what they are doing.
 		if pth := "/" + strings.TrimLeft(r.URL.Path, "/"); pth != "/readyz" && pth != "/healthz" && pth != "/livez" {
 			if isLocal(r) {
-				klog.V(2).Infof("Loopback request to %q (user agent %q) before server is ready. This client probably does not watch /readyz and might get inconsistent answers.", r.URL.Path, r.UserAgent())
+				if !isKubeApiserverLoopBack(r) {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:119]: you could change it to if `isLocal(r) && !isKubeApiserverLoopBack(r)`

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go [hunk 2]
```diff
@@ -143,3 +145,7 @@ func isLocal(req *http.Request) bool {
 
 	return false
 }
+
+func isKubeApiserverLoopBack(req *http.Request) bool {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:149]: based on https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/release-openshift-origin-installer-e2e-remote-libvirt-s390x-4.6/1308979535697416192/artifacts/e2e-remote-libvirt/pods/openshift-kube-apiserver_kube-apiserver-ci-op-tjzfzjcn-486b5-lzwlc-master-1_kube-apiserver.log

at least the following operators are also using loopback:
- `cluster-kube-controller-manager-operator`
- `cluster-kube-apiserver-operator`


do you know why?
would we like to put them on a white list?

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go [hunk 3]
```diff
@@ -143,3 +145,7 @@ func isLocal(req *http.Request) bool {
...
+	return strings.HasPrefix(req.UserAgent(), "kube-apiserver/")
```
[the remark this answers is not in the document]
Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:150]: I don't think.

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:119]: no, then the else case is hit and it shouldn't.

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:149]: that's a follow-up.

It is the DR sidecar afaik. But before whitelisting that we have to change the user agent there. We do not want to whitelist kcm-o or ka-o as a whole.

File: staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go (hunk 1 above)
Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go:119]: ah, ic, correct.

**Pull request merged**